[Song Bingxi] iP

Propene-L opened the pull request:
# Veneto
> “Your mind is for having ideas, not holding them.” – David Allen [source](https://dansilvestre.com/productivity-quotes)

Veneto frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ _SUPER_ FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2324s1.github.io/website/schedule/week4/project.html).
1. double-click it.
1. add your tasks.
1. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [x] Managing tasks
- [x] Managing deadlines
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Propene-L committed:
add level-0

Propene-L committed:
added Level 1

Propene-L committed:
add MyList.java

Propene-L committed:
added level 2

Propene-L committed:
add Task.java

Propene-L committed:
changing MyList to a Task List

Propene-L committed:
added level 3

Propene-L committed:
add Deadline.java

Propene-L committed:
add Event.java

Propene-L committed:
add toDo.java

Propene-L committed:
add some comments to Task.java

Propene-L committed:
modified MyList.java

Propene-L committed:
added level 4

Propene-L committed:
bug fixed

Propene-L committed:
added DanException

Propene-L committed:
add level 5

Propene-L committed:
added level 6

Propene-L committed:
used enums

Propene-L committed:
bug fixed after using enums

Propene-L committed:
input.txt modified

zacwong2151 commented on src/main/java/Dan.java:
```diff
@@ -0,0 +1,172 @@
+public class Dan {
+
+    private final static String greets = "\nDan: \n";
+    private static MyList tasks = new MyList(100);
```
perhaps a class name like TaskList would be more intuitive than MyList?

zacwong2151 commented on src/main/java/Dan.java:
```diff
@@ -0,0 +1,172 @@
+        System.out.println(
+                greets +
+                " 好啦，帮你擦掉了一条任务哦：\n " + removedTask +
+                "\n 现在还剩下" + tasks.size() + "项任务哦！\n"
```
only english is recommended for coding!

zacwong2151 commented on src/main/java/Dan.java:
```diff
@@ -0,0 +1,172 @@
+import java.util.Arrays;
+import java.util.Scanner;
+
+enum TASK_TYPE {TODO, DEADLINE, EVENT}
```
Good job on setting your ToDos as enums!

zacwong2151 commented on src/main/java/Dan.java:
```diff
@@ -0,0 +1,172 @@
+            command = texts[0].toLowerCase();
+            switch (command) {
+                case "bye":
+                    break a;
```
perhaps it would be better if you can extract these 'instructions' out to a Ui class

zacwong2151 commented on src/main/java/Dan.java:
```diff
@@ -0,0 +1,172 @@
+            try {
+            text = new Scanner(System.in).nextLine();
+            texts = text.split(" ");
+            command = texts[0].toLowerCase();
```
can consider extracting the commands to a Parser or Command class

zacwong2151 reviewed (commented):
Looks good to me except for some parts
1. everything should be written in English!
2. some parts could be extracted out into Ui, Parser classes
3. do consider adding a description for your PR!

starrylight99 reviewed (commented):
pokemon exception, try to be more specific in catching your exceptions :)

Propene-L committed:
updated deadline.java with functions

Propene-L committed:
updated event.java with funcs

Propene-L committed:
updated task.java with funcs

Propene-L committed:
updated todo.java with functions

Propene-L committed:
updated test-ui-test

Propene-L committed:
add level 7

Propene-L committed:
Merge branch 'branch-Level-7'

Propene-L committed:
change field deadline into type "LocalDate"

Propene-L committed:
change field "start", "end" into type "LocalDate"

Propene-L committed:
added level-8

Propene-L committed:
bugs fixed

Propene-L committed:
updated input.txt

Propene-L committed:
Merge branch 'branch-Level-8'

hcs1203 commented on src/main/java/Dan.java:
```diff
@@ -0,0 +1,238 @@
+            "mark [TASK_ID]", "unmark [TASK_ID]",
+            "list", "bye"
+    };
+    private final static String savePath = "./data/dan.txt";
```
Use capital letters for constants

hcs1203 commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,31 @@
+    protected LocalDate deadline;
+
+    // toString
+
```
Delete empty lines

hcs1203 commented on src/main/java/MyList.java:
```diff
@@ -0,0 +1,18 @@
+
+    @Override
+    public String toString() {
+        StringBuffer rt = new StringBuffer();
```
name should be clearer for the reader

hcs1203 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,47 @@
+        return this.description + "," + (isDone ? 1 : 0);
+    }
+
+    // Constructor
```
Place constructor on top

hcs1203 commented on src/main/java/ToDo.java:
```diff
@@ -0,0 +1,25 @@
+    @Override
+    public String saveToString() {
+        return "toDo," + super.saveToString();
+    }
```
Unless specifically using both functions, can combine

hcs1203 reviewed (commented):
Good work overall. Code is mostly clear to read and follow along. Can show more care in naming.

Propene-L committed:
added AddCommand.java

Propene-L committed:
renamed MyTask.java -> TaskList.java

Propene-L committed:
added ListCommand

Propene-L committed:
added ExitCommand

Propene-L committed:
added Command.java as abstract class

Propene-L committed:
make package dan/task

Propene-L committed:
mk package dan.exception

Propene-L committed:
added toString() to AddCommand

Propene-L committed:
added toString to DeleteCommand

Propene-L committed:
added MarkCommand.java

Propene-L committed:
fixed bugs after packaging DanException

Propene-L committed:
added Storage.java

Propene-L committed:
replaced DanException.java

Propene-L committed:
added Ui.java

Propene-L committed:
added Parser.java

Propene-L committed:
added A-OOP

Propene-L committed:
created DanOperateException

Propene-L committed:
created DanStorageException

Propene-L committed:
Fixed Bugs after adding more Exceptions

Propene-L committed:
fixed bugs in task.java after added more Exceptions

Propene-L committed:
modified showError()

Propene-L committed:
fixed bugs in Add command

Propene-L committed:
added A-OOP

Propene-L committed:
Merge tag 'A-MoreOOP'

Propene-L committed:
added A-Packages

Propene-L committed:
Merge tag 'A-Packages'

Propene-L committed:
Added Parser.java

Propene-L committed:
apply parser in ui

Propene-L committed:
modified AddCommand to apply Parser

Propene-L committed:
modified Command

Propene-L committed:
modified Delete

Propene-L committed:
modified Exit

Propene-L committed:
Modified List

Propene-L committed:
Modified Mark

Propene-L committed:
added to package

Propene-L committed:
bugs fixed After added packages

Propene-L committed:
Merge commit 'ed6d4d25bd4e34461919ffe785a6f10f4c087552'

Propene-L committed:
renamed to Veneto

Propene-L committed:
Further renamed to Veneto

Propene-L committed:
change showError static

Propene-L committed:
added ToDOTest

Propene-L committed:
Added EventTest

Propene-L committed:
added UiTest

Propene-L committed:
Added AddCommandTest(Will be filled later)

Propene-L committed:
Added MarkCommandTest(Will be filled later)

Propene-L committed:
Added VenetoTest(Will be filled later)

Propene-L committed:
Merge tag 'A-JUnit'

Propene-L committed:
put Veneto.java to package veneto

Propene-L committed:
rename constants to ALL_CAPITAL

Propene-L committed:
Merge tag 'A-Jar'

Propene-L committed:
renamed DanException to VenetoException

Propene-L committed:
add JavaDoc to Veneto.java

Propene-L committed:
add JavaDoc to AddCommand

Propene-L committed:
add JavaDoc to Command

Propene-L committed:
add JavaDoc to DeleteCommand

Propene-L committed:
add JavaDoc to ExitCommand

Propene-L committed:
add JavaDoc to ListCommand

Propene-L committed:
add JavaDoc to MarkCommand

Propene-L committed:
add JavaDoc to Parser

Propene-L committed:
add JavaDoc to Storage

Propene-L committed:
add JavaDoc to Deadline

Propene-L committed:
add JavaDoc to Event

Propene-L committed:
add JavaDoc to Task

Propene-L committed:
add JavaDoc to TaskList

Propene-L committed:
add JavaDoc to ToDo

Propene-L committed:
add JavaDoc to Ui

Propene-L committed:
Modified UiTest

Propene-L committed:
fixed bugs after renameing DanExceptions to VenetoExceptions

Propene-L committed:
added FindCommand

Propene-L committed:
modified Task to support FindCommand

Propene-L committed:
modified Parser to support FindCommand

Propene-L committed:
modified Ui to support FindCommand

Propene-L committed:
added test file(will be filled later)

Propene-L committed:
changed naming

Propene-L committed:
Merge branch 'branch-A-JavaDoc'

Conflicts resolved

# Conflicts:
#	src/main/java/veneto/storage/Storage.java
#	src/main/java/veneto/task/Task.java
#	src/main/java/veneto/ui/Ui.java

Propene-L committed:
Merge branch 'A-CodingStandard'

Conflicts resolved

# Conflicts:
#	src/main/java/veneto/ui/Ui.java

Propene-L committed:
Merge tag 'level-9'

Conflicts resolved

# Conflicts:
#	src/main/java/veneto/parser/Parser.java
#	src/main/java/veneto/task/Task.java

Propene-L committed:
further resolved merge conflicts

Propene-L committed:
add Constants to avoid magic numbers

Propene-L committed:
make getType() non-static

Propene-L committed:
modified Ui

Propene-L committed:
added tests

Kevin-Liusx commented on README.md:
```diff
@@ -13,12 +13,14 @@ Prerequisites: JDK 11, update Intellij to the most recent version.
+      \  \/  /  |   _______||          ||   _______|    |  |    |  |    |  |
+       \    /   |  |_______ |   |\     ||  |_______     |  |    |  |____|  |
+        \__/    |__________||___|  \___||__________|    |__|    |__________|
+   
```
Nice logo!

Kevin-Liusx reviewed (commented):
I think how you added JavaDoc comments for all your methods and variables

johnnythesnake12 commented on src/main/java/veneto/parser/Parser.java:
```diff
@@ -0,0 +1,95 @@
+        Task newTask = null;
+        String description;
+        switch (funcId) {
+            case 1:
```
Hi perhaps instead of calling it case 1,2,3 you could name the cases such that it is easier for others to understand which cases this are?

Propene-L committed:
added JavaFX dependencies

Propene-L committed:
added Launcher.java

Propene-L committed:
replace storage file

Propene-L committed:
added MainWindow.fxml

Propene-L committed:
added DialogBox.fxml

Propene-L committed:
added MainWindow.java

Propene-L committed:
added DialogBox.java

Propene-L committed:
added images

Propene-L committed:
added Main.java

Propene-L committed:
added Launcher

Propene-L committed:
changed Veneto: chat() -> getResponse()

Propene-L committed:
modified Ui

Propene-L committed:
Merge branch 'level-10'

Propene-L committed:
added a new jar file

Propene-L committed:
Renaming fields to improve code quality.

renaming filePath to storagePath in Veneto.

Propene-L committed:
removed redundant try-catch to improve readability

Propene-L committed:
modified mark() in Task.java

added markTask() and unmarkTask() to avoid deep nesting and
improved readability

Propene-L committed:
added constants to avoid magic numbers

added EXIT_SUFFIX to show operated command is ExitCommand
added INCORRECT_ERROR to show unidentified error
modifed COMMAND and make it looks better

added three methods to improve readability

added showOperationError() to deal with VenetoOperationException
added showStorageError() to deal with VenetoStorageException
added showInvalidError() to deal with unexpected exceptions

Propene-L committed:
fixed bugs in MainWindow.java

Propene-L committed:
Merge pull request #2 from Propene-Dan/A-CodeQuality

merging A-codequality to master

Propene-L committed:
enable assertions

Propene-L committed:
added constants STORAGE_DIR and STORAGE_PATH in Storage.java

Propene-L committed:
added assertion for invariants in Parser

Propene-L committed:
added assertion for invariants in Storage

Propene-L committed:
added assertion for invariants in Ui

Propene-L committed:
Merge branch 'A-Assertions'

# Conflicts:
#	src/main/java/veneto/Veneto.java
#	src/main/java/veneto/ui/Ui.java

Conflicts resolved

Propene-L committed:
modified type of VenetoStorageException when no storage found

Propene-L committed:
added isDuplicated() to check duplicate task when adding newTask to TaskList

Propene-L committed:
added javaDoc to AddCommand.java

Propene-L committed:
added equals() to Task.java

Propene-L committed:
added equals() to ToDo.java

Propene-L committed:
added equals() and detect DateTimeParseException to Deadline.java

Propene-L committed:
added equals() and detect DateTimeParseException to Event.java

Propene-L committed:
catch DateTimeParseException in Parser.java

Propene-L committed:
add response when found duplicates

Propene-L committed:
Merge branch 'C-DetectDuplicates'

Propene-L committed:
Add files via upload

Ui.png

Propene-L committed:
Update README.md

Propene-L committed:
i dont know

Propene-L committed:
test

Propene-L committed:
Merge branch 'lalala'

Propene-L committed:
test2

Propene-L committed:
Merge branch 'lalala'

Propene-L committed:
lalala 1

Propene-L committed:
lalala 2

Propene-L committed:
oi

Propene-L committed:
Merge branch 'lalala1'

# Conflicts:
#	src/main/java/veneto/Test.java

Propene-L committed:
no message

Propene-L committed:
Merge commit 'e0e610e5684b75b6ba5b255576fac0083f21d26a'
